### PR TITLE
Scraper/pdf

### PR DESCRIPTION
**Pull request by HarrisonGregg**

### CODE REVIEW

**Comment by wearpants** [comment]
interesting - what is this used for?

**Comment by HarrisonGregg** [comment]
There are a few links in the spreadsheet to files like [this](http://www.vernoncollege.edu/Resources/Instructional%20Information%202010%20-%202011/ACCT%20CL.pdf) that have direct links to syllabus files embedded in PDFs.

**Comment by wearpants** [comment]
cool, please give it a docstring (one sentence is fine) otherwise looks good

File: osp_scraper/spiders/pdf.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+import scrapy
+import PyPDF2 as pyPdf
```
Comment (jodizzle) [inline osp_scraper/spiders/pdf.py]: This means that `PyPDF2` should be added to `requirements.txt`, right?

File: osp_scraper/spiders/pdf.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from io import BytesIO
+
+from ..spiders.CustomSpider import CustomSpider
+
+class PDFSpider(CustomSpider):
+    name = "pdf"
+
+    def start_requests(self):
+        for url in PDFSpider.start_urls:
```
Comment (jodizzle) [inline osp_scraper/spiders/pdf.py]: Maybe you should use `parse` instead?

File: osp_scraper/spiders/pdf.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                ann = p["/Annots"]
+                for a in ann:
+                    o = a.getObject()
+                    if "/A" in o:
+                        A = o["/A"]
+                        if "/URI" in A:
+                            uri = A["/URI"]
+                            if type(uri) is pyPdf.generic.ByteStringObject:
+                                uri = uri.decode('utf-8').replace('\x00','')
+                            yield((uri, uri))
```
Comment (jodizzle) [inline osp_scraper/spiders/pdf.py]: Are two levels of parentheses necessary?

File: osp_scraper/spiders/pdf.py (hunk 2 above)
Comment (HarrisonGregg) [inline osp_scraper/spiders/pdf.py]: Done.

File: osp_scraper/spiders/pdf.py (hunk 3 above)
Comment (HarrisonGregg) [inline osp_scraper/spiders/pdf.py]: Fixed.

File: osp_scraper/spiders/pdf.py (hunk 1 above)
Comment (HarrisonGregg) [inline osp_scraper/spiders/pdf.py]: Good call. Done.

**Comment by HarrisonGregg** [comment]
@jodizzle Look good now?

File: osp_scraper/spiders/pdf.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        pdf = pyPdf.PdfFileReader(BytesIO(response.body))
+        pgs = pdf.getNumPages()
+
+        for page_num in range(pgs):
+            page = pdf.getPage(page_num)
+
+            annotations = page.get("/Annots", [])
+            for annotation in annotations:
+                annot_object = annotation.getObject()
+
```
Comment (jodizzle) [inline osp_scraper/spiders/pdf.py]: What's up with the newline spacing you're doing here?  I guess you're trying to separate sections of the looping process?

It's not big deal, just looks a bit weird.

File: osp_scraper/spiders/pdf.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from io import BytesIO
+
+import PyPDF2 as pyPdf
+
+from ..spiders.CustomSpider import CustomSpider
+
+class PDFSpider(CustomSpider):
```
Comment (jodizzle) [inline osp_scraper/spiders/pdf.py]: @wearpants Mentioned that you should provide a docstring.

Comment (HarrisonGregg) [inline osp_scraper/spiders/pdf.py]: Whoops, thought I already did that.

**Pull request merged**